### ISSUE TITLE
DB not found on DS213+

### ISSUE THREAD

**Issue by zaadstra** [007revad/Synology_HDD_db#10]
Hi,

I tested the script (I cut off the writing stuff part) and it errored on the DB file part.  I've read another issue and ran some of the commands to collect info.

System: DS213+ with DSM 6.2.4 25556-6

get_key_value /etc/synoinfo.conf unique   (also /etc.defaults/synoinfo.conf)
synology_qoriq_213+

cat /proc/cmdline
.... syno_hw_version=DS213pv10 .....

cat /proc/sys/kernel/syno_hw_version
DS213pv10-j

cat /etc.defaults/synoinfo.conf | grep disk_com
support_disk_compatibility="yes"

LOG PART:

/sys/block/sda
Model:    'WD40EFRX-68N32N0'
Firmware: '82.0'
/sys/block/sdb
Model:    'WD40EFRX-68N32N0'
Firmware: '82.0'
/sys/block/sdq
Model:    'STORAGE DEVICE'
Firmware: '9910'
/sys/block/sdr
Model:    'STORAGE DEVICE'
Firmware: '9910'
/sys/block/sdr
add:    'STORAGE DEVICE,9910'
add:    'WD40EFRX-68N32N0,82.0'
HDD/SSD models found: 2
STORAGE DEVICE,9910
WD40EFRX-68N32N0,82.0

No M.2 drives found

/sys/block/sdr
ERROR /var/lib/disk-compatibility/ds213pv10_host.db not found!

In /var/lib/disk-compatibility/ there are two DB files related to DS213+ (among others).  They are recently updated because a fresh DSM install and hitting the Update now button for Drive Database in Storage Manager.
The used harddrives are present in the files:

ds213+_host.db = ds213+_host.db.new: {"success":1,"list":[ ....
,{"model":"WD40EFRX-68N32N0","firmware":"82.00A82","default_unverified":0,"rec_intvl":[3]},

ds213+_host_v7.db = ds213+_host_v7.db.new: {"disk_compatbility_info":{ ....
,"WD40EFRX-68N32N0":{"82.00A82":{"compatibility_interval":[{"compatibility":"not_support","not_yet_rolling_status":"not_support","fw_dsm_update_status_notify":false,"barebone_installable":true}]},

I cannot find which of those two files actually is used.  I don't know what STORAGE DEVICE is, this is not present in both files.

A long time ago these drives were new and installed with DSM 5.2 (probably).  This was migrated to 6.x and now a fresh install.

At this install to my surprise the Storage Manager complained about the disks "Incompatible with Synology NAS: Drive x".
After 'continue' the drives installed and now in the Storage Manager I cannot see any complaints anymore about this.
So I'm wondering if I should put effort in changing the DB file anyway.  I also could hack it in manually.

**Comment by 007revad** [comment]
There's a number of things that caught my attention: 
1. DS213pv10
2. DS213pv10-j
3. Model: 'STORAGE DEVICE'
4. "default**_unverified**"
5. "compatibility":"**not_support**"
6. "not_yet_rolling_status":"**not_support**"
7. ds213+_host_v7.db

1 and 2 - I've seen models that return "-j" after their model number even though they are "+" models. But I've not seen "pv10" before. It looks like the kernel, and maybe mainboard, is used for DS213+, DS213, DS213j (and DS213air which I'd never heard of before).

Does the following command return the same DS213pv10-j
`get_key_value /etc/synoinfo.conf upnpmodelname`

On my DS1812+ `cat /proc/sys/kernel/syno_hw_version` returns DS1812+ even when both synoinfo.conf files are changed to 1813+ so `cat /proc/sys/kernel/syno_hw_version` seemed the most reliable method... until now.

I'll probably parse the 213+ from "synology_qoriq_213+" and remove the DS from "DS213pv10-j" and if the results don't match (213+ != 213pv10-j) I'll add the "DS" and "213+" together.

3 - 'STORAGE DEVICE' is strange. I need to look into this.

4, 5 and 6 - It looks like I might need to edit existing entries in the db file, instead of just checking the drive exists in the db file.

7 - I thought ds213+_host_v7.db only existed on NASes running DSM 7. But maybe it exists on models that can update to DSM 7, even if they are still using DSM 6. Because you are using DSM 6 it *should* be using the ds213+_host.db file.

**Comment by 007revad** [comment]
> So I'm wondering if I should put effort in changing the DB file anyway.

On a DS213+ with DSM 6 you shouldn't see any drive compatibility warnings apart from the initial warning when initializing the drive. So yes, I wouldn't bother changing the db file.

**Comment by zaadstra** [comment]
good morning ;-)

Results of the queries:

get_key_value /etc/synoinfo.conf upnpmodelname
DS213+

./script.sh
unique:    213+
model:     DS213pv10-j
modelnum:  213
modeltype: DS

NAS is a DS213+

7. The DS213+ cannot get a DSM7 update.

Since the complaints are gone, and I can't check changes to the DB I make, I think I just leave it.

**Comment by zaadstra** [comment]
Also wondering about the
Model:    'STORAGE DEVICE'
Firmware: '9910'

Only thing I can think of are the two connected USB disks, but they are powered off (power supply disconnected).

**Comment by 007revad** [comment]
You are correct. The two 'STORAGE DEVICE' are USB drives. The USB ports from the Synology would be providing enough power to detect them (if the USB cables are still connected).

I just finished testing if the two 'STORAGE DEVICE' could have been USB drives. The 'sdq' and 'sdr' was also a clue because the 220+ only has 2 drive bays (which would be sda and sdb).

So I plugged a USB drive into my Synology and sdq appeared. I also found how to check if a drive is a removable drive and ignore it in my script. So thank you for bringing this to my attention.
